deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/LSTMHelpers.java

```diff
@@ -249,6 +251,15 @@ static public FwdPassReturn activateHelper(final Layer layer, final NeuralNetCon
                 currHiddenUnitActivations.muliColumnVector(timeStepMaskColumn);
                 currentMemoryCellState.muliColumnVector(timeStepMaskColumn);
             }
+           // boolean layerNormalization = true;
+            if (useLayerNormalization) {
+                Number stdev = currHiddenUnitActivations.stdNumber();
```

Comment:
See equations 3 and 4 in the referenced preprint (https://arxiv.org/pdf/1607.06450.pdf). From what I understand, the mean and standard deviations are for all activations, not column-wise. They are scalars in the equations. I think that is why the normalization can be applied to a mini-batch of size 1 (as claimed in the preprint as an advantage over batch normalization).

Regarding masks, it is possible handling masks could improve the results, or not. I have tried this on a problem where we use masking, and found adequate normalization of the activations without explicitly handling the masks. There is also no information in the preprint about masking, so I think the implementation is accurate in this respect.